src/audio/codec_adapter/codec/generic.c

```diff
@@ -263,11 +263,13 @@ int module_process(struct comp_dev *dev)
 	md->state = MODULE_PROCESSING;
 
 	ret = md->ops->process(dev);
-	if (ret)
+	if (ret && ret != -ENOSPC) {
```

Comment:
@ranj063 can you provide more details why it should be skipped?

Reply:
@mwasko ENOSPC is marker telling us that process() wasn't actually called due to not enough data in the input buffer or not not enough free space in the output buffer. This is not an error per se. It is just a state in the pipeline where we should move to next iteration in order to get more data at the input or consume more data at the output.